src/common/filesystembase.cpp

```diff
@@ -433,15 +435,75 @@ bool FileSystem::remove(const QString &fileName, QString *errorString)
         setFileReadOnly(fileName, false);
     }
 #endif
-    QFile f(fileName);
-    if (!f.remove()) {
-        if (errorString) {
-            *errorString = f.errorString();
+#ifdef Q_OS_UNIX
+    if(_moveToTrash){
+        Q_UNUSED(errorString);
+        qDebug() << "moving" << fileName << "to trash";
+        if (!moveToTrash(fileName)) {
+            return false;
+        }
+    }
+    if(!_moveToTrash) {
```

Comment:
Please use else, and put the brace outside the #if so that you don't need another #if for the closing brace

```
    } else
#endif
    {
```